src/kudu/consensus/consensus_queue.cc

```diff
@@ -1051,6 +1062,51 @@ void PeerMessageQueue::PromoteIfNeeded(TrackedPeer* peer, const TrackedPeer& pre
   }
 }
 
+bool PeerMessageQueue::BasicChecksOKToTransferUnlocked(
+    const TrackedPeer& peer, RaftPeerPB **peer_pb_ptr) {
```

Comment:
nit: Looks like this method is doing more than just checks, it also populated peer_pb_ptr, maybe rename it?